packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -643,14 +670,77 @@ export const getNodesWithStyles = ( tree, blockSelectors ) => {
 
 			if ( node?.variations ) {
 				const variations = {};
-				Object.keys( node.variations ).forEach( ( variation ) => {
-					variations[ variation ] = pickStyleKeys(
-						node.variations[ variation ]
-					);
-				} );
+
```

Comment:
The following changes mostly follow those in the PHP class for processing block style variations into nodes that can be later turned into actual styles.